components/BidderBio.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from 'react';
+import Text, { textStyle } from './Text';
+import { Wallet } from '../server/api/types';
+import { useAccount, useSignMessage } from 'wagmi';
+import { trpc } from '../utils/trpc';
+import { formatAddress, checkIsAuthor } from '../utils/utils';
+
+export default function BidderBio({ bidder }: { bidder: Wallet }) {
+  const { address } = useAccount();
+  const { signMessageAsync } = useSignMessage();
+
+  const mutation = trpc.private.insertBio.useMutation();
+
+  const [bio, setBio] = useState(bidder.bio || '');
+  const [message, setMessage] = useState<{ text: string; isSuccess: boolean } | null>(null);
```

Comment:
You don't need this state, you can infer it from `mutation`: `mutation.error`, `mutation.isSuccess`, etc.